src/Datadog.Trace/Services/RandomIdProvider.cs

```diff
@@ -0,0 +1,29 @@
+using System;
+using Datadog.Trace.Interfaces;
+
+namespace Datadog.Trace.Services
+{
+    internal class RandomIdProvider : IIdProvider
+    {
+        private readonly IRandomProvider _randomProvider;
+
+        internal RandomIdProvider(IRandomProvider randomProvider)
+        {
+            _randomProvider = randomProvider ?? throw new ArgumentNullException(nameof(randomProvider));
+        }
+
+        public static RandomIdProvider Instance { get; } = new RandomIdProvider(SimpleDependencyFactory.RandomProvider());
+
+        public ulong GetUInt63Id()
+        {
+            // From https://stackoverflow.com/a/677390
+            var buffer = new byte[sizeof(ulong)];
+
```

Comment:
Just a formatting nit. I tend to use more empty lines than most C# developers (at least more than @bmermet and @dd-caleb), but even this seems a bit excessive to me.

Reply:
I agree, but our linter thing keeps yelling at me...probably only on some of these and not all, but once it started yelling I just started hitting enter.... I'll go see if maybe I can figure out the specifics, but I'm with you here if we can get that thing to stop erroring...